### PR TITLE
CLN: replacing str.format with f-strings in several files. #29547

### LINKED ISSUE #29547: Replace old string formatting syntax with f-strings
Since we no longer support python 3.5, we can now use the new f-strings instead of the old ```.format()``` ( and obviously the ```%``` formatting).

#### Notes:
* Don't forget to link this issue in your pull request's body message , simply paste this ```https://github.com/pandas-dev/pandas/issues/29547``` in your pull request's body message.

* If any of your changed files are related to #29886 , please make sure to link your pull request to that issue as well, simply paste this ```https://github.com/pandas-dev/pandas/issues/29886``` in your pull request's body message as well.

* Please comment what you are planning to work on, so we won't do double work.

* If a file/files that should be marked as done, is not marked, please comment letting me know.

---

To check what files still needs to be fixed in the ```pandas``` directory:
```
grep -l -R '%s'  --include=*.{py,pyx} pandas/
grep -l -R '%d' --include=*.{py,pyx} pandas/
grep -l -R '\.format(' --include=*.{py,pyx} pandas/
```

All of the above can also be used as a one liner:
```
grep -l -R -e '%s' -e '%d' -e '\.format(' --include=*.{py,pyx} pandas/
```

##### Tip:
If you want to see the line number of the occurrence, replace the ```-l``` with ```-n```
for example:
```
grep -n -R '%s' --include=*.{py,pyx} pandas/
```

---

### The current list is:

- [ ] pandas/compat/pickle_compat.py

- [ ] pandas/_config/config.py

- [ ] pandas/core/arrays/datetimelike.py

- [ ] pandas/core/arrays/datetimes.py

- [ ] pandas/core/arrays/integer.py

- [ ] pandas/core/arrays/period.py

- [ ] pandas/core/computation/pytables.py

- [ ] pandas/core/config_init.py

- [ ] pandas/core/frame.py

- [ ] pandas/core/generic.py

- [ ] pandas/core/groupby/generic.py

- [ ] pandas/core/groupby/groupby.py

- [ ] pandas/core/indexes/base.py

- [ ] pandas/core/indexes/multi.py

- [ ] pandas/core/indexes/range.py

- [ ] pandas/core/ops/docstrings.py

- [ ] pandas/core/ops/\_\_init__.py

- [ ] pandas/core/reshape/merge.py

- [ ] pandas/core/tools/datetimes.py

- [ ] pandas/io/formats/css.py

- [ ] pandas/io/formats/excel.py

- [ ] pandas/io/formats/format.py

- [ ] pandas/io/formats/html.py

- [ ] pandas/io/formats/info.py

- [ ] pandas/io/formats/latex.py

- [ ] pandas/io/formats/printing.py

- [ ] pandas/io/formats/style.py

- [ ] pandas/io/parsers.py

- [ ] pandas/io/pytables.py

- [ ] pandas/io/sas/sas_xport.py

- [ ] pandas/io/stata.py

- [ ] pandas/_libs/tslibs/c_timestamp.pyx

- [ ] pandas/_libs/tslibs/frequencies.pyx

- [ ] pandas/_libs/tslibs/parsing.pyx

- [ ] pandas/_libs/tslibs/period.pyx

- [ ] pandas/_libs/tslibs/strptime.pyx

- [ ] pandas/_libs/tslibs/timedeltas.pyx

- [ ] pandas/plotting/_matplotlib/converter.py

- [ ] pandas/tests/arrays/categorical/test_operators.py

- [ ] pandas/tests/arrays/test_datetimelike.py

- [ ] pandas/tests/dtypes/test_dtypes.py

- [ ] pandas/tests/extension/base/setitem.py

- [ ] pandas/tests/frame/test_constructors.py

- [ ] pandas/tests/frame/test_missing.py

- [ ] pandas/tests/frame/test_to_csv.py

- [ ] pandas/tests/groupby/aggregate/test_other.py

- [ ] pandas/tests/indexes/datetimes/test_date_range.py

- [ ] pandas/tests/indexes/datetimes/test_datetime.py

- [ ] pandas/tests/indexes/datetimes/test_formats.py

- [ ] pandas/tests/indexes/datetimes/test_partial_slicing.py

- [ ] pandas/tests/indexes/interval/test_constructors.py

- [ ] pandas/tests/indexes/interval/test_interval.py

- [ ] pandas/tests/indexes/multi/test_format.py

- [ ] pandas/tests/indexes/period/test_formats.py

- [ ] pandas/tests/indexes/test_base.py

- [ ] pandas/tests/indexes/timedeltas/test_timedelta.py

- [ ] pandas/tests/indexing/test_categorical.py

- [ ] pandas/tests/indexing/test_coercion.py

- [ ] pandas/tests/io/excel/test_openpyxl.py

- [ ] pandas/tests/io/excel/test_writers.py

- [ ] pandas/tests/io/formats/test_format.py

- [ ] pandas/tests/io/formats/test_printing.py

- [ ] pandas/tests/io/formats/test_style.py

- [ ] pandas/tests/io/formats/test_to_csv.py

- [ ] pandas/tests/io/formats/test_to_html.py

- [ ] pandas/tests/io/formats/test_to_latex.py

- [ ] pandas/tests/io/parser/test_compression.py

- [ ] pandas/tests/io/parser/test_encoding.py

- [ ] pandas/tests/io/parser/test_header.py

- [ ] pandas/tests/io/parser/test_parse_dates.py

- [ ] pandas/tests/io/parser/test_usecols.py

- [ ] pandas/tests/io/test_html.py

- [ ] pandas/tests/io/test_sql.py

- [ ] pandas/tests/io/test_stata.py

- [ ] pandas/tests/reductions/test_reductions.py

- [ ] pandas/tests/reshape/test_concat.py

- [ ] pandas/tests/reshape/test_melt.py

- [ ] pandas/tests/scalar/period/test_period.py

- [ ] pandas/tests/scalar/timedelta/test_timedelta.py

- [ ] pandas/tests/scalar/timestamp/test_constructors.py

- [ ] pandas/tests/series/indexing/test_numeric.py

- [ ] pandas/tests/series/indexing/test_take.py

- [ ] pandas/tests/series/indexing/test_where.py

- [ ] pandas/tests/series/methods/test_rename.py

- [ ] pandas/tests/series/test_api.py

- [ ] pandas/tests/series/test_constructors.py

- [ ] pandas/tests/series/test_datetime_values.py

- [ ] pandas/tests/series/test_repr.py

- [ ] pandas/tests/test_strings.py

- [ ] pandas/tests/tools/test_to_datetime.py

- [ ] pandas/tests/tseries/holiday/test_calendar.py

- [ ] pandas/tests/tseries/holiday/test_holiday.py

- [ ] pandas/tests/tslibs/test_parsing.py

- [ ] pandas/tests/util/test_assert_frame_equal.py

- [ ] pandas/tseries/frequencies.py

- [ ] pandas/util/_decorators.py

- [ ] pandas/util/_test_decorators.py

- [ ] pandas/util/_validators.py

- [ ] pandas/_version.py

---

#### NOTE:

The list may change as files are moved/renamed constantly.

---

Inhereted files and commands from this [PR](https://github.com/pandas-dev/pandas/issues/16130).

### PR DESCRIPTION
**Pull request by jlamborn324**
- [ ] xref #29547 
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/test_style.py [hunk 1]
```diff
@@ -1215,11 +1215,7 @@ def test_highlight_max(self):
 
     def test_export(self):
         f = lambda x: "color: red" if x > 0 else "color: blue"
-        g = (
-            lambda x, y, z: "color: {z}".format(z=z)
-            if x > 0
-            else "color: {z}".format(z=z)
-        )
+        g = lambda x, y, z: f"color: {z}" if x > 0 else f"color: {z}"
```
Comment (charlesdong1991) [inline pandas/tests/io/formats/test_style.py]: not related to the issue, maybe `y` can be removed here?

**Review by charlesdong1991 — COMMENTED** [review]
very nice! one minor comment (not related to the issue though)

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -1610,7 +1606,7 @@ def test_to_string_small_float_values(self):
 
         result = df.to_string()
         # sadness per above
-        if "{x:.4g}".format(x=1.7e8) == "1.7e+008":
+        if f"{1.7e8:.4g}" == "1.7e+008":
```
Comment (simonjayhawkins) [inline pandas/tests/io/formats/test_format.py]: can `_three_digit_exp` function be used here instead?
```suggestion
        if _three_digit_exp():
```

File: pandas/tests/io/formats/test_format.py [hunk 3]
```diff
@@ -1939,7 +1935,7 @@ def test_repr_html_float(self):
             ).set_index("idx")
             reg_repr = df._repr_html_()
             assert ".." not in reg_repr
-            assert "<td>{val}</td>".format(val=str(40 + h)) in reg_repr
+            assert f"<td>{str(40 + h)}</td>" in reg_repr
```
Comment (simonjayhawkins) [inline pandas/tests/io/formats/test_format.py]: is str needed?
```suggestion
            assert f"<td>{40 + h}</td>" in reg_repr
```

File: pandas/tests/io/formats/test_format.py [hunk 4]
```diff
@@ -1951,8 +1947,8 @@ def test_repr_html_float(self):
             ).set_index("idx")
             long_repr = df._repr_html_()
             assert ".." in long_repr
-            assert "<td>{val}</td>".format(val="31") not in long_repr
-            assert "{h} rows ".format(h=h) in long_repr
+            assert f"<td>{str(31)}</td>" not in long_repr
```
Comment (simonjayhawkins) [inline pandas/tests/io/formats/test_format.py]: is f-string needed?

```suggestion
            assert "<td>31</td>" not in long_repr
```

File: pandas/tests/io/formats/test_format.py [hunk 5]
```diff
@@ -1240,7 +1238,7 @@ def test_wide_repr(self):
             set_option("display.expand_frame_repr", False)
             rep_str = repr(df)
 
-            assert "10 rows x {c} columns".format(c=max_cols - 1) in rep_str
+            assert f"10 rows x {(max_cols - 1)} columns" in rep_str
```
Comment (simonjayhawkins) [inline pandas/tests/io/formats/test_format.py]:
```suggestion
            assert f"10 rows x {max_cols - 1} columns" in rep_str
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jlamborn324 for the PR. generally lgtm. a few minor suggestions

File: pandas/tests/io/formats/test_style.py (hunk 1 above)
Comment (jlamborn324) [reply pandas/tests/io/formats/test_style.py]: Good idea! I'll look into taking care of it

File: pandas/tests/io/formats/test_format.py [hunk 6]
```diff
@@ -711,7 +709,7 @@ def format_func(x):
 
     def test_to_string_with_formatters_unicode(self):
         df = DataFrame({"c/\u03c3": [1, 2, 3]})
-        result = df.to_string(formatters={"c/\u03c3": lambda x: "{x}".format(x=x)})
+        result = df.to_string(formatters={"c/\u03c3": lambda x: f"{x}"})
```
Comment (jbrockmendel) [inline pandas/tests/io/formats/test_format.py]: can this just be `str(x)`?

Comment (jlamborn324) [reply pandas/tests/io/formats/test_format.py]: Good call. Thank you for your help!

File: pandas/tests/io/formats/test_format.py [hunk 7]
```diff
@@ -711,7 +709,7 @@ def format_func(x):
 
     def test_to_string_with_formatters_unicode(self):
         df = DataFrame({"c/\u03c3": [1, 2, 3]})
-        result = df.to_string(formatters={"c/\u03c3": lambda x: "{x}".format(x=x)})
+        result = df.to_string(formatters={"c/\u03c3": lambda x: str(x)})
```
Comment (simonjayhawkins) [inline pandas/tests/io/formats/test_format.py]:
```suggestion
        result = df.to_string(formatters={"c/\u03c3": str})
```

**Review by simonjayhawkins — APPROVED** [review]
@jlamborn324 lgtm

**Comment by datapythonista** [comment]
Thanks for the clean up @jlamborn324

**Pull request merged**